test/realtime_box_best_effort_tests.cpp

```diff
@@ -122,15 +124,18 @@ TEST(RealtimeBoxBestEffort, assignment_operator)
 }
 TEST(RealtimeBoxBestEffort, typecast_operator)
 {
-  RealtimeBoxBestEffort box(DefaultConstructable{.a = 100, .str = ""});
+  DefaultConstructable data_construct;
+  data_construct.a = 100;
+  data_construct.str = "";
+  RealtimeBoxBestEffort box(data_construct);
 
   // Use non RT access
   DefaultConstructable data = box;
 
   EXPECT_EQ(data.a, 100);
 
   // Use RT access -> returns std::nullopt if the mutex could not be locked
-  std::optional<DefaultConstructable> rt_data_access = box;
+  std::optional<DefaultConstructable> rt_data_access = box.tryGet();
```

Comment:
I am a bit puzzled why this doesn't work. I don't remember seeing a warning before on that line. (Which is now escalated to an error)

Reply:
Maybe is this warning not included in `-Wall -Wextra`?

Reply:
I believe what @christophfroehlich did here is the right approach. The problem is that the box is of type `RealtimeBoxBestEffort<T>` and we are initializing std::optional<T> with this, so when using flag `-Werror=conversion` this fails